### PR TITLE
[V3.5.0 deferred pipeline] fix vulkan crash problem

### PR DESCRIPTION
**Pull request by star-e**
Fixed deferred pipeline vulkan crash problem

Changelog:
 * Fixed vulkan crash problem by implementing depth stencil texture as input attachment.
 * Updated deferred lighting shader to support depth as input attachment.
 * Disabled framebuffer fetch. FBF is not correctly implemented in gles3.
 * Updated Planar Shadow Queue to render shadow in subpass 1, which is the lighting pass. Currently it is wrongly rendered in GBuffer pass.

Issues:
 * https://github.com/cocos/3d-tasks/issues/10948

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

**Review by GengineJS — APPROVED** [review]

**Review by minggo — APPROVED** [review]

File: native/cocos/renderer/pipeline/deferred/GbufferStage.cpp [hunk 1]
```diff
@@ -205,6 +205,9 @@ void GbufferStage::render(scene::Camera *camera) {
             static_cast<uint>(pipeline->getWidth() * shadingScale),
             static_cast<uint>(pipeline->getHeight() * shadingScale),
         };
+        if (_device->getGfxAPI() == gfx::API::VULKAN) {
+            depthTexInfo.usage |= gfx::TextureUsageBit::INPUT_ATTACHMENT;
+        }
         data.depth = builder.create(DeferredPipeline::fgStrHandleOutDepthTexture, depthTexInfo);
```
Comment (hana-alice) [inline native/cocos/renderer/pipeline/deferred/GbufferStage.cpp:211]: 这里后端应该是能通过beginrenderPass inputs来判断的？

**Review by hana-alice — APPROVED** [review]

File: native/cocos/renderer/pipeline/deferred/GbufferStage.cpp (hunk 1 above)
Comment (star-e) [reply native/cocos/renderer/pipeline/deferred/GbufferStage.cpp:211]: 资源相关的设置，我觉得显式标注更明确些，交给用户负责，所以不去推导该有的状态。
这里更好的写法是通过gfx::Feature::INPUT_ATTACHMENT_BENEFIT来判断是否标注INPUT_ATTACHMENT。
这方面没有完全梳理过，所以先针对vk开启，之后再完善。

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by pandamicro** [comment]
Ignoring eslint result due to ci issue

**Pull request merged**